pypeman/contrib/http.py

```diff
@@ -1,33 +1,82 @@
 import asyncio
 import ssl
 import warnings
-
-from pypeman import endpoints, channels, nodes, message
+import socket
 
 import aiohttp
 from aiohttp import web
 
+from pypeman import endpoints, channels, nodes, message
+from pypeman.errors import PypemanParamError
+
+
 class HTTPEndpoint(endpoints.BaseEndpoint):
     """
     Endpoint to receive HTTP connection from outside.
 
-    :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
-        `client_max_size`
     """
 
-    def __init__(self, adress=None, address='127.0.0.1', port='8080', loop=None, http_args=None):
+    def __init__(
+            self, adress=None, 
+            address=None, 
+            port=None, 
+            loop=None, 
+            http_args=None, 
+            url=None, 
+            sock=None,
+            reuse_port=False,
+            ):
+        """
+            :param http_args: dict to pass as **kwargs** to aiohttp.Application for example for
+                `client_max_size`
+            :param reuse_port: bool if true then the listening port specified in the url parameter) 
+                will be shared with other processes on same port
+            :param sock: socket object already bound and configured
+        """
         super().__init__()
+        default_url = 'localhost:8080'
```

Comment:
I think this variable is unuseful, can you remove it ?

Reply:
It's just use once but I think it's more readable to have default value here.

Reply:
I think the default value should be described in documentation for users and in place where it's used in the code, but if you insist i will not insist again ;-)